### ISSUE TITLE
TypeScript type support for .json method

### ISSUE THREAD

**Issue by unknownsoldier08** [tncrazvan/sveltekit-sse#31]
Is there any example on how to correctly type the .json() method using TypeScript?

**Comment by tncrazvan** [comment]
Hi @unknownsoldier08 ,

I just released version `0.8.9` with better type hinting support for `.json()`, please to the latest version.

You should now get proper documentation in your editor.
![image](https://github.com/tncrazvan/sveltekit-sse/assets/6891346/916c9b19-dfd9-4f14-a094-5ae50c4e1b76)

A full example using typescript would look like this

```svelte
<script lang="ts">
  import { source } from 'sveltekit-sse';
  const channel = source('/events').select('notification');
  const notification = channel.json<{ content: string }>();
</script>

{$notification?.content}
```

And with error management

```svelte
<script lang="ts">
  import { source } from 'sveltekit-sse';
  const channel = source('/events').select('notification');
  const notification = channel.json<{ content: string }>(function onError({ previous, error }) {
    console.error(error);
    return previous;
  });
</script>

{$notification?.content}
```

Let me know if this is what you're looking for.

**Comment by unknownsoldier08** [comment]
That works, thank you!
